src/Components/Alert/Hooks/__tests__/AlertProvider.jest.tsx

```diff
@@ -0,0 +1,121 @@
+import { MockEnvironment, createMockEnvironment } from "relay-test-utils"
+import { useTracking } from "react-tracking"
+import { graphql } from "react-relay"
+
+import { AlertProvider } from "Components/Alert/AlertProvider"
+import { CreateAlertButton } from "Components/Alert/Components/CreateAlertButton"
+import { AlertProviderTestPage } from "Components/Alert/Hooks/__tests__/Utils/AlertProviderTestPage"
+import { useAuthDialog } from "Components/AuthDialog"
+import { setupTestWrapper } from "DevTools/setupTestWrapper"
+import { useSystemContext } from "System/SystemContext"
+import { flushPromiseQueue } from "DevTools/flushPromiseQueue"
+
+jest.unmock("react-relay")
+jest.mock("react-tracking")
+
+jest.mock("Components/AuthDialog/useAuthDialog")
+jest.mock("Utils/Events", () => ({
+  postEvent: jest.fn(),
+}))
+jest.mock("System/useFeatureFlag", () => ({
+  useFeatureFlag: jest.fn(() => true),
+}))
+jest.mock("System/useSystemContext")
+
+describe("AlertProvider", () => {
+  let relayEnv: MockEnvironment = createMockEnvironment()
+  let trackEvent: jest.Mock
+
+  const { getWrapper } = setupTestWrapper({
+    Component: ({ artist }: any) => (
+      <AlertProvider initialCriteria={{ artistIDs: [artist.internalID] }}>
+        <CreateAlertButton />
+      </AlertProvider>
+    ),
+    query: graphql`
+      query AlertProviderTestQuery @raw_response_type @relay_test_operation {
+        artist(id: "artist-id") {
+          internalID
+        }
+      }
+    `,
+  })
+
+  beforeAll(() => {
+    trackEvent = jest.fn()
+    ;(useTracking as jest.Mock).mockImplementation(() => ({ trackEvent }))
+    ;(useSystemContext as jest.Mock).mockImplementation(() => {
+      return {
+        isLoggedIn: true,
+        relayEnvironment: relayEnv,
+      }
+    })
+    ;(useAuthDialog as jest.Mock).mockImplementation(() => {
+      return { showAuthDialog: jest.fn() }
+    })
+  })
+
+  afterEach(() => {
+    relayEnv.mockClear()
+  })
+
+  it("opens the create alert modal with initial criteria", async () => {
```

Comment:
@tam-kis This should give us a good base to continue iterating on. Will add some additional specs exercising the filter option in a follow-up PR.